core/app/models/spree/address.rb

```diff
@@ -34,25 +32,16 @@ class Address < Spree::Base
       end
     end
 
-    alias_attribute :first_name, :firstname
-    alias_attribute :last_name, :lastname
-    alias_attribute :full_name, :name
-
     DB_ONLY_ATTRS = %w(id updated_at created_at)
     TAXATION_ATTRS = %w(state_id country_id zipcode)
-    LEGACY_NAME_ATTRS = %w(firstname lastname full_name)
+    LEGACY_NAME_ATTRS = %w(firstname lastname)
```

Comment:
Do we really need to keep this constant? I saw it's used in `as_json` but I can't get why, can you help me understand?

Reply:
I've kept it to make explicit why we remove them from the json hash.
Since they're still on the db they are added by default to any attributes representation, including the json one, but we want to keep them private and avoid that they creep outside the model polluting API responses and breaking equality checks.

Reply:
Got it, it's clear now. I wasn't thinking that the list was coming magically from db columns. Have you considered using Rails' `ignore_columns` for this? There's also a [similar example](https://github.com/rails/rails/blob/b80645003796690b76aef5c289923db543d84673/activerecord/test/models/developer.rb#L25) in Rails specs. 😆

Reply:
Yes, I thought about it but we still need those columns to be loaded, our internal implementation of `name` accessor still rely on them, I plan to remove them all togheter in the next PR 👍